### PR TITLE
chore: bump golang to 1.23 and gomodules.xyz/logs to v0.0.7

### PR DESCRIPTION
**Pull request by mainred**
Current guard image built has several CVEs. We want to bump the golang and toolchain, bundled together for simplicity, to 1.23.1 to for 1.23.0 has been reported to have CVEs as well [here](https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/2095)

A higher golang version without bumping gomodules.xyz/logs to v0.0.7 will bring the following issue
```
panic: flag stderrthreshold set at /src/vendor/gomodules.xyz/logs/lib.go:41 before being defined

goroutine 1 [running]:
flag.(*FlagSet).Var(0xc0000f2150, {0x21b2018, 0x2fa1184}, {0x1bab4bd, 0xf}, {0x1c41be1, 0x8a})
        /usr/local/go/src/flag/flag.go:1031 +0x314
k8s.io/klog/v2.InitFlags.func1(0xc0002b25a0?)
        /src/vendor/k8s.io/klog/v2/klog.go:439 +0x31
flag.(*FlagSet).VisitAll(0xc000762560?, 0xc00074de40)
        /usr/local/go/src/flag/flag.go:458 +0x42
k8s.io/klog/v2.InitFlags(0x16?)
        /src/vendor/k8s.io/klog/v2/klog.go:438 +0x45
gomodules.xyz/logs.Init(0xc0001c3b08, 0x0)
        /src/vendor/gomodules.xyz/logs/lib.go:61 +0x25
main.main()
        /src/main.go:28 +0x31
```

```
trivy image --format table --exit-code  1 --ignore-unfixed --vuln-type  os,library --severity  CRITICAL,HIGH,MEDIUM,LOW,UNKNOWN mcr.microsoft.com/oss/kubeguard/guard:v0.16.11

Total: 14 (UNKNOWN: 0, LOW: 1, MEDIUM: 10, HIGH: 2, CRITICAL: 1)

┌───────────────────────────────────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬──────────────────────────────────────────────────────────────┐
│                Library                │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version
  │                            Title                             │
├───────────────────────────────────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼──────────────────────────────────────────────────────────────┤
│ github.com/golang-jwt/jwt/v4          │ CVE-2024-51744 │ LOW      │ fixed  │ v4.2.0            │ 4.5.1
  │ golang-jwt: Bad documentation of error handling in           │
│                                       │                │          │        │                   │
  │ ParseWithClaims can lead to potentially...                   │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-51744                   │
├───────────────────────────────────────┼────────────────┼──────────┤        ├───────────────────┼─────────────────┼──────────────────────────────────────────────────────────────┤
│ github.com/hashicorp/go-retryablehttp │ CVE-2024-6104  │ MEDIUM   │        │ v0.7.2            │ 0.7.7
  │ go-retryablehttp: url might write sensitive information to   │
│                                       │                │          │        │                   │
  │ log file                                                     │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-6104                    │
├───────────────────────────────────────┼────────────────┼──────────┤        ├───────────────────┼─────────────────┼──────────────────────────────────────────────────────────────┤
│ stdlib                                │ CVE-2024-24790 │ CRITICAL │        │ 1.20.14           │ 1.21.11, 1.22.4 │ golang: net/netip: Unexpected behavior from Is methods for   │
│                                       │                │          │        │                   │
  │ IPv4-mapped IPv6 addresses                                   │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-24790                   │
│                                       ├────────────────┼──────────┤        │                   ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2023-45288 │ HIGH     │        │                   │ 1.21.9, 1.22.2
  │ golang: net/http, x/net/http2: unlimited number of           │
│                                       │                │          │        │                   │
  │ CONTINUATION frames causes DoS                               │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2023-45288                   │
│                                       ├────────────────┤          │        │                   ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-34156 │          │        │                   │ 1.22.7, 1.23.1
  │ encoding/gob: golang: Calling Decoder.Decode on a message    │
│                                       │                │          │        │                   │
  │ which contains deeply nested structures...                   │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-34156                   │
│                                       ├────────────────┼──────────┤        │                   ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2023-45289 │ MEDIUM   │        │                   │ 1.21.8, 1.22.1
  │ golang: net/http/cookiejar: incorrect forwarding of          │
│                                       │                │          │        │                   │
  │ sensitive headers and cookies on HTTP redirect...            │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2023-45289                   │
│                                       ├────────────────┤          │        │                   │
  ├──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2023-45290 │          │        │                   │
  │ golang: net/http: golang: mime/multipart: golang:            │
│                                       │                │          │        │                   │
  │ net/textproto: memory exhaustion in                          │
│                                       │                │          │        │                   │
  │ Request.ParseMultipartForm                                   │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2023-45290                   │
│                                       ├────────────────┤          │        │                   │
  ├──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-24783 │          │        │                   │
  │ golang: crypto/x509: Verify panics on certificates with an   │
│                                       │                │          │        │                   │
  │ unknown public key algorithm...                              │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-24783                   │
│                                       ├────────────────┤          │        │                   │
  ├──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-24784 │          │        │                   │
  │ golang: net/mail: comments in display names are incorrectly  │
│                                       │                │          │        │                   │
  │ handled                                                      │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-24784                   │
│                                       ├────────────────┤          │        │                   │
  ├──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-24785 │          │        │                   │
  │ golang: html/template: errors returned from MarshalJSON      │
│                                       │                │          │        │                   │
  │ methods may break template escaping                          │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-24785                   │
│                                       ├────────────────┤          │        │                   ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-24789 │          │        │                   │ 1.21.11, 1.22.4 │ golang: archive/zip: Incorrect handling of certain ZIP files │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-24789                   │
│                                       ├────────────────┤          │        │                   ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-24791 │          │        │                   │ 1.21.12, 1.22.5 │ net/http: Denial of service due to improper 100-continue     │
│                                       │                │          │        │                   │
  │ handling in net/http                                         │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-24791                   │
│                                       ├────────────────┤          │        │                   ├─────────────────┼──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-34155 │          │        │                   │ 1.22.7, 1.23.1
  │ go/parser: golang: Calling any of the Parse functions        │
│                                       │                │          │        │                   │
  │ containing deeply nested literals...                         │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-34155                   │
│                                       ├────────────────┤          │        │                   │
  ├──────────────────────────────────────────────────────────────┤
│                                       │ CVE-2024-34158 │          │        │                   │
  │ go/build/constraint: golang: Calling Parse on a "// +build"  │
│                                       │                │          │        │                   │
  │ build tag line with...                                       │
│                                       │                │          │        │                   │
  │ https://avd.aquasec.com/nvd/cve-2024-34158                   │
└───────────────────────────────────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴──────────────────────────────────────────────────────────────┘
```

### CODE REVIEW

**Comment by mainred** [comment]
Kindly requesting @weinong's help to take a look.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module go.kubeguard.dev/guard
 
-go 1.18
+go 1.23.1
```
Comment (sozercan) [inline go.mod:3]: this updates minimum version to be go 1.23.1. By just building with latest go, you'll get rid of the stdlib CVEs. Unless the project is using go 1.23 features, you don't need to bump this.

Comment (mainred) [reply go.mod:3]: 1.23 has already been used in [CI](https://github.com/kubeguard/guard/pull/394/files), so I think to align all golang version should be ok?

**Comment by mainred** [comment]
CI is green, ping @weinong to take a look.

**Review by weinong — APPROVED** [review]
LGTM

**Pull request merged**